builder/vsphere/common/step_keep_cdrom.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:generate packer-sdc struct-markdown
+//go:generate packer-sdc mapstructure-to-hcl2 -type KeepCDRomConfig
+
+package common
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/packer-plugin-sdk/multistep"
+	packersdk "github.com/hashicorp/packer-plugin-sdk/packer"
+	"github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/driver"
+)
+
+type KeepCDRomConfig struct {
+	// Keep a number of CD-ROM devices. Range: 1-4.
+	// If set to 0, `keep_cdrom` is ignored and the step is skipped.
+	// When set to a value in the range, `remove_cdrom` is ignored and
+	// the CD-ROM devices are kept without any attached media.
+	KeepCdrom int32 `mapstructure:"keep_cdrom"`
```

Comment:
It looks like the number of cdroms you can keep don't correlate to the number of cdroms attached by the cdrom configurations. Is that okay? 

That said the name is confusing as I would've expected it to be a bool and not an int type. If it has no relation to the number of cdroms attached it is a bit more confusting to me. Should this be `keep_cdroms`

```go
       // Keep a number of CD-ROM devices with no attached media. Range: 1-4.
	// If set to 0, `keep_cdroms` is ignored and the step is skipped.
	// When set to a value in the range, `remove_cdrom` is ignored and
	// the CD-ROM devices are kept without any attached media.
	KeepCdrom int32 `mapstructure:"keep_cdrom"`
```

Does it matter which CDRoms are kept?

Reply:
No. All existing cd-rom devices are ejected and removed. Then the count of cd-rom devices are added using the same controller type defined in the image spec. The devices have to mount some media based on the AddCdrom implemenation but then proptly eject it (used the vmtools iso on the host.)